treon/treon.py

```diff
@@ -107,24 +107,40 @@ def print_test_collection(notebooks):
 
 
 def get_notebooks_to_test(args):
-    path = args['PATH'] or os.getcwd()
-    result = []
-
-    if os.path.isdir(path):
-        LOG.info('Recursively scanning %s for notebooks...', path)
-        path = os.path.join(path, '')  # adds trailing slash (/) if it's missing
-        glob_path = path + '**/*.ipynb'
-        result = glob.glob(glob_path, recursive=True)
-    elif os.path.isfile(path):
-        if path.lower().endswith('.ipynb'):
-            LOG.debug('Testing notebook %s', path)
-            result = [path]
-        else:
-            sys.exit('{path} is not a Notebook'.format(path=path))
+    path = pathlib.Path(args['PATH'] or os.getcwd())
+
+    if path.is_file():
+        if path.suffix == '.ipynb':
+            return [path.as_posix()]
+        sys.exit("{path} is not a notebook".format(path=path))
+    elif path.is_dir():
+        LOG.info("Recursively scanning %s for notebooks...", path)
+        ignored = build_ignore_list(path)
+        notebooks = filter(lambda nb: nb not in ignored, path.glob('**/*.ipynb'))
+        return (nb.as_posix() for nb in notebooks)
     else:
-        sys.exit('{path} is not a valid path'.format(path=path))
-
-    if not result:
-        sys.exit('No notebooks to test in {path}'.format(path=path))
-
-    return result
+        sys.exit("{path} is not a valid path".format(path=path))
+
+    sys.exit("No notebooks to test in {path}".format(path=path))
```

Comment:
This would never get executed even when all notebooks are ignored. You might want to switch back to the original version of `result = [x, y, z]` notebooks. That way we can test for result being present in every case.